### PR TITLE
QL: Remove the unnecessary `DateTimeFunction.dateTimeFormat()`

### PR DESCRIPTION
**Pull request by palesz**
Removes the `DateTimeFunction.dateTimeFormat()` and the 
`TranslatorHandler.dateTimeFormat()` methods that were
called, but had no effect on the translated queries.

One of the examples, when this happens, is when one side of the
`BinaryComparison` is a function. The `BinaryComparison` at the
end turns into a `ScriptQuery`, but in the intermediate steps the
`ExpressionTranslators` try to translate it to a `RangeQuery`.

Follows #68783

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/SqlTranslatorHandler.java
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/SqlTranslatorHandler.java:39]: Shouldn't it be: `querySupplier.get()`?

**Review by bpintea — COMMENTED** [review]
Besides Marios' comment - and maybe apart from a missing test, if feasible - it LGTM. Good spotting on unused/-sable formats.

**Review by costin — CHANGES_REQUESTED** [review]
Good catch. However this PR needs to be broken into two - it's a bit more tedious but cleaner.
It looks to me there's a PR for removing date format and another one for adding lazy translation.
The latter in particular has subtle implications and as others have pointed out, it looks like we don't have complete testing in that area - hence the breakage.

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/SqlTranslatorHandler.java
Comment (palesz) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/SqlTranslatorHandler.java:39]: Thanks, good catch! Seems there is no test, good time to add one.

**Comment by palesz** [comment]
> Good catch. However this PR needs to be broken into two - it's a bit more tedious but cleaner.
> It looks to me there's a PR for removing date format and another one for adding lazy translation.
> The latter in particular has subtle implications and as others have pointed out, it looks like we don't have complete testing in that area - hence the breakage.

Makes sense, will break it down into two PRs.

**Comment by palesz** [comment]
Removed the introduction of lazyness from this PR, will resubmit once this PR is merged to avoid the conflicts.

**Review by bpintea — APPROVED** [review]

**Review by matriv — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**
